extensions/iidm/include/powsybl/iidm/extensions/iidm/DiscreteMeasurement.hpp

```diff
@@ -0,0 +1,197 @@
+
+    /**
+     * Get ID of the measurement if it exists. It is optional (can be empty).
+     */
+    const std::string& getId() const;
+
+    /**
+     * Return property value associated to the given property name if it exists, else return an empty string.
+     */
+    std::string getProperty(const std::string& name) const;
+
+    /**
+     * Get all the property names with values applied to this measurement.
+     */
+    stdcxx::const_range<std::string> getPropertyNames() const;
+
+    /**
+     * Get the tap changer the measurement is applied on if necessary (see {@link TapChanger}).
+     * Must be an invalid reference if the measurement is not applied to a tap changer. Throw an exception if the extended equipment is not a transformer.
+     * Must be a valid reference if the measurement is not applied to a tap changer, else throw an exception.
+     */
+    stdcxx::optional<TapChanger> getTapChanger() const;
+
+    /**
+     * Get the type of the measurement (specify what it is measured, see {@link Type}).
+     */
+    const Type& getType() const;
+
+    /**
+     * Get value as boolean if the type of measured discrete value is BOOLEAN (see {@link ValueType}).
+     * Else throw an exception.
+     */
+    bool getValueAsBoolean() const;
+
+    /**
+     * Get value as int if the type of measured discrete value is INT (see {@link ValueType}).
+     * Else throw an exception.
+     */
+    int getValueAsInt() const;
+
+    /**
+     * Get value as String if the type of measured discrete value is STRING (see {@link ValueType}).
+     * Else throw an exception.
+     */
+    std::string getValueAsString() const;
+
+    /**
+     * Get the type of measured discrete value (int, boolean or string, see {@link ValueType}).
+     */
+    const ValueType& getValueType() const;
+
+    /**
+     * Get validity status of the measurement.
+     * If it is true (i.e. the measurement is valid), the discrete measured value can not be null.
+     */
+    bool isValid() const;
+
+    /**
+     * Put a given property value associated with a given property name.
+     */
+    DiscreteMeasurement& putProperty(const std::string& name, const std::string& property);
+
+    /**
+     * Remove the measurement from the equipment.
+     */
+    void remove();
+
+    /**
+     * Remove the property value associated with the given property name if it exists. Else, do nothing.
+     */
+    DiscreteMeasurement& removeProperty(const std::string& name);
+
+    /**
+     * Set validity status of the measurement.
+     * If it is true (i.e. the measurement is valid), the discrete measured value can not be null.
+     */
+    DiscreteMeasurement& setValid(bool valid);
+
+    /**
+     * Set the discrete measured value and set the value type as STRING (see {@link ValueType}).
+     * Can not be empty if the measurement is valid.
+     */
+    DiscreteMeasurement& setValue(const std::string& value);
+
+    DiscreteMeasurement& setValue(const char* value);
```

Comment:
This one is here to avoid implicit cast to boolean version when calling with a char*